csrc/multidevice/lower_communication.cpp

```diff
@@ -196,7 +196,7 @@ void lowerToReduceScatter(
     std::vector<Communication*>& comms) {
   const DeviceMesh& mesh = input_tv->getDeviceMesh();
   auto reduction_axis = output_tv->getReductionAxis().value();
-  auto scattered_axis = getShardedAxis(output_tv);
+  auto scattered_axis = getShardedAxis(output_tv, ParallelType::DIDx);
```

Comment:
ok, however if the sharded dimension is split, then `scatted_axis` is not valid here, right?

Reply:
I can't think of an immediate problem and https://github.com/NVIDIA/Fuser/pull/3504 apparently works fine. Could be incidental and I'm happy to hear what you think is problematic.

Reply:
In your case, 
```
tv1:
  logical: [r{i0}, i{i1}]
  allocation: [r{i0}, i{i1/d}, iDID{d}]
```

getShardedLogicalAxis will return 0, the tensor axis being sharded. This is correct because the output at::Tensor for will be of shape [i1/d] and indeed axis 0 is the sharded dimension. 

Caveat: With https://github.com/NVIDIA/Fuser/commit/7cf238402fdbd0656f26577088128418e7433f6d, DID'ing an inner split is disallowed by code. So the above case will actually throw an exception. But what I said should be correct after we lift that limitation.